common/persistence/client/factory.go

```diff
@@ -267,8 +338,12 @@ func (f *factoryImpl) NewVisibilityManager() (p.VisibilityManager, error) {
 	if ds.ratelimit != nil {
 		result = p.NewVisibilityPersistenceRateLimitedClient(result, ds.ratelimit, f.logger)
 	}
-	if visConfig != nil && visConfig.EnableSampling() {
-		result = p.NewVisibilitySamplingClient(result, visConfig, f.metricsClient, f.logger)
+	if visibilityConfig.EnableDBVisibilitySampling() {
+		result = p.NewVisibilitySamplingClient(result, &p.SamplingConfig{
```

Comment:
Fields in `ResourceConfig` can be `nil`, but we are not doing any checks on the caller side or callee side.

Reply:
What kind of checking do you think we should have? I am open to add more. There is one checking in the factoryImpl which would return nil when both read and write mode are nil.

Reply:
Something similar to line 324 before invoking the functions

Reply:
Ah I see. Added the nil check here.